### PR TITLE
docs: Update LB-IPAM `allowFirstLastIPs` documentation

### PR DESCRIPTION
**Pull request by dylandreimerink**
We changed the default value of the `allowFirstLastIPs` in Cilium CiliumLoadBalancerIPPool objects. This wasn't reflected in the documentation. This commit updates the documentation to match reality.

### CODE REVIEW

**Review by youngnick — APPROVED** [review]

File: Documentation/network/lb-ipam.rst [hunk 1]
```diff
@@ -64,25 +64,26 @@ An IP pool can have multiple blocks of IPs. A block can be specified with CIDR
...
+blocks specified with ``.spec.blocks[].start`` and ``.spec.blocks[].stop``.
 
 .. warning::
 
-  In v1.15, ``.spec.allowFirstLastIPs`` defaults to ``no``. This will change to
-  ``yes`` in v1.16. Please set this field explicitly if you rely on the field
-  being set to ``no``.
+  In v1.15, ``.spec.allowFirstLastIPs`` defaults to ``No``. This has changed to
+  ``Yes`` in v1.16. Please set this field explicitly if you rely on the field
+  being set to ``No``.
```
Comment (learnitall) [inline Documentation/network/lb-ipam.rst:86]: Would you mind adding this to the v1.16 upgrade notes?

**Review by learnitall — CHANGES_REQUESTED** [review]
Great! One small request but otherwise LGTM. Thanks!

File: Documentation/network/lb-ipam.rst (hunk 1 above)
Comment (dylandreimerink) [reply Documentation/network/lb-ipam.rst:86]: Its already there https://github.com/cilium/cilium/blob/v1.16/Documentation/operations/upgrade.rst?plain=1#L383-L386

Comment (learnitall) [reply Documentation/network/lb-ipam.rst:86]: Darn missed that somehow, thanks!

**Review by learnitall — APPROVED** [review]
Thanks!

**Pull request merged**